x/pairing/keeper/pairing.go

```diff
@@ -173,24 +189,66 @@ func (k Keeper) getProjectStrictestPolicy(ctx sdk.Context, project projectstypes
 	}
 
 	if !projectstypes.CheckChainIdExistsInPolicies(chainID, policies) {
-		return 0, 0, "", 0, fmt.Errorf("chain ID not found in any of the policies")
+		return 0, 0, "", 0, 0, []string{}, fmt.Errorf("chain ID not found in any of the policies")
 	}
 
 	geolocation := k.CalculateEffectiveGeolocationFromPolicies(policies)
 
 	providersToPair := k.CalculateEffectiveProvidersToPairFromPolicies(policies)
 	if providersToPair == uint64(math.MaxUint64) {
-		return 0, 0, "", 0, fmt.Errorf("could not calculate providersToPair value: all policies are nil")
+		return 0, 0, "", 0, 0, []string{}, fmt.Errorf("could not calculate providersToPair value: all policies are nil")
 	}
 
 	sub, found := k.subscriptionKeeper.GetSubscription(ctx, project.GetSubscription())
 	if !found {
-		return 0, 0, "", 0, fmt.Errorf("could not find subscription with address %s", project.GetSubscription())
+		return 0, 0, "", 0, 0, []string{}, fmt.Errorf("could not find subscription with address %s", project.GetSubscription())
 	}
 	allowedCU := k.CalculateEffectiveAllowedCuPerEpochFromPolicies(policies, project.GetUsedCu(), sub.GetMonthCuLeft())
 
 	projectToPair := project.Index
-	return geolocation, providersToPair, projectToPair, allowedCU, nil
+
+	selectedProvidersMode, selectedProvidersList := k.CalculateEffectiveSelectedProviders(policies)
+
+	return geolocation, providersToPair, projectToPair, allowedCU, selectedProvidersMode, selectedProvidersList, nil
+}
+
+func (k Keeper) CalculateEffectiveSelectedProviders(policies []*projectstypes.Policy) (projectstypes.PolicySelectedProvidersModeEnum, []string) {
+	selectedProvidersModeList := []projectstypes.PolicySelectedProvidersModeEnum{}
+	selectedProvidersList := [][]string{}
+	for _, p := range policies {
+		selectedProvidersModeList = append(selectedProvidersModeList, p.SelectedProvidersMode)
+		if p.SelectedProvidersMode == projectstypes.Policy_EXCLUSIVE || p.SelectedProvidersMode == projectstypes.Policy_MIXED {
+			if len(p.SelectedProviders) != 0 {
```

Comment:
do we need this check?

Reply:
yes. this check makes empty list to be "all providers". without it, the intersection will be empty